File: FastBuy-main/Modulo_Client_CSharp/Modulo_C_sharp/Prodotto.cs
namespace Modulo_C_sharp
{
    public class Prodotto
    {
        private string cod;
        private string categoria;
        private string nome;
        private float prezzo;
        private int quantita;

        public Prodotto(string cod, string categoria, string nome, float prezzo, int quantita)
        {
            Cod = cod;
            Categoria = categoria;
            Nome = nome;
            Prezzo = prezzo;
            Quantita = quantita;
        }

        public string Cod { get => cod; set => cod = value; }
        public string Categoria { get => categoria; set => categoria = value; }
        public string Nome { get => nome; set => nome = value; }
        public float Prezzo { get => prezzo; set => prezzo = value; }
        public int Quantita { get => quantita; set => quantita = value; }
    }
}

File: FastBuy-main/Modulo_Server_Cpp/prova3/Prodotto.cpp
#include "Prodotto.h"

Prodotto::Prodotto(string cod, string categoria, string nome, float prezzo, int quantita) {
	this->cod = cod;
	this->categoria = categoria;
	this->nome = nome;
	this->prezzo = prezzo;
	this->quantita = quantita;
}

string Prodotto::getCod() {
	return cod;
}

string Prodotto::getCategoria() {
	return categoria;
}

string Prodotto::getNome() {
	return nome;
}

float Prodotto::getPrezzo() {
	return prezzo;
}

int Prodotto::getQuantita() {
	return quantita;
}

ostream& operator << (ostream& os, const Prodotto& p) {
	os << p.cod + " " + p.categoria + " " + p.nome + " " << p.prezzo << " " << p.quantita;
	return os;
}